### PR TITLE
feat(client): related articles

### PR DESCRIPTION
**Pull request by DDonochVA**

### CODE REVIEW

File: libs/blog/articles/feature-article/src/lib/article-details/article-details.component.html [hunk 1]
```diff
@@ -30,11 +30,13 @@ <h2 id="article-title" class="flex text-[40px] font-bold">
...
-        <!--  @todo replace to similar-articles    -->
-        <!--                <al- class="block min-h-[288px]" />-->
+
+      @defer (on viewport(view); on timer(5s)) {
+        <section>
+          <al-related-articles [id]="articleDetails().id" />
+        </section>
       } @placeholder {
-        <div class="h-[288px]"></div>
+        <div></div>
```
Comment (mateuszbasinski) [inline libs/blog/articles/feature-article/src/lib/article-details/article-details.component.html:39]: do we need this empty div?

File: libs/blog-bff/articles/api/src/lib/wp-posts.ts [hunk 2]
```diff
@@ -16,6 +16,19 @@ export class WpPosts {
     });
   }
 
+  getRelatedPosts(
+    id: string,
+    query: Record<string, string | number>,
+  ): Promise<WPResponse<WPPostDto[]>> {
+    return this._wpClient.get<WPPostDto[]>(`related/${id}`, {
+      ...query,
+      limit: '2',
```
Comment (DamianBrzezinskiHoA) [inline libs/blog-bff/articles/api/src/lib/wp-posts.ts:25]: Should it be hardcoded ?

File: libs/blog/articles/feature-related-articles/src/lib/related-articles.component.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [RelatedArticleListStore],
+  imports: [UiArticleCardComponent],
+})
+export class RelatedArticlesComponent implements OnInit {
+  readonly id = input.required<number>();
+
+  readonly store = inject(RelatedArticleListStore);
```
Comment (DamianBrzezinskiHoA) [inline libs/blog/articles/feature-related-articles/src/lib/related-articles.component.ts:32]: We can add `protected`

**Review by mateuszbasinski — APPROVED** [review]

File: libs/blog/articles/feature-related-articles/src/lib/related-articles.component.ts (hunk 3 above)
Comment (DDonochVA) [reply libs/blog/articles/feature-related-articles/src/lib/related-articles.component.ts:32]: skipping for now

**Pull request merged**